File: 0650/solution.py
from math import ceil, sqrt


class Solution:
    def minSteps(self, n: int) -> int:
        factor = 2
        res = 0
        root = ceil(sqrt(n))
        while factor <= root:
            while n % factor == 0:
                res += factor
                n //= factor
            factor += 1
        return res if n == 1 else res + n

File: 0650/solution.java
class Solution {
    public int minSteps(int n) {
        int res = 0;
        int root = (int) Math.ceil(Math.sqrt(n));
        int factor = 2;
        while (factor <= root) {
            while (n % factor == 0) {
                res += factor;
                n /= factor;
            }
            factor++;
        }

        return n > 1 ? res + n : res;
    }
}

// time O(sqrt(n)log(sqrt(n)))
// space O(1)
